### PR TITLE
Feature Create frontend UI for Vehicle calculator

### LINKED ISSUE #41: As a user, I want to be able to determine my carbon emissions from my driving
**__User Story Description__**
As a user
I want to be able to determine my carbon emissions from my driving
So I can further inform my CO2 estimate

**__Steps to Follow (optional)__**
- [ ] Create form on '/electricity' page
- [ ] Communicate with Developer on Issue #18 for questions to put in UI
- [ ] Form on submit should send correct data to backend
- [ ] Add additional steps as necessary

**__Additional Considerations__**
Keep in mind the backend version of this ticket #18

### PR DESCRIPTION
**Pull request by TiciaD**
- added more types to vehicle schema
- added endpoints for vehicle makes and models to router
- Created Questionaire, Response and Form components for Vehicle page
- added react-select-search package for searchable dropdown component

Closes #41

### CODE REVIEW

File: src/components/Vehicle/VehicleForm.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    }
+  }, [selectedVehicleMake, models.isLoading, models.data]);
+
+  const isVehicleMakesDisabled = vehicleMakes?.length == 0;
+  const isVehicleModelsDisabled = vehicleModels?.length == 0;
+  const isSubmitDisabled =
+    !selectedVehicleMake || !selectedVehicleModel || !distance;
+
+  const handleVehicleMakeChange = (
+    value: SelectedOptionValue | SelectedOptionValue[] | string
```
Comment (ialej001) [inline src/components/Vehicle/VehicleForm.tsx:56]: A smarter solution to what I suggested the other day :)

File: src/server/router/vehicle.router.ts [hunk 2]
```diff
@@ -16,4 +18,46 @@ export const vehicleRouter = createRouter().mutation(
...
+      const response = await fetch(
+        // Query keeps firing on component load so added a fallback value to keep from getting invalid input errors
+        carbonVehicleModelsRequest(input ? input : '4c1e16e1-7967-4394-b3cb-15f4577dffa1')
+      );
+      let vehicleOptions: VehicleModelSearch[] = []
+
+      return response.json().then((data) => {
+        const vehicles = data as VehicleModelDataResponse[];
+        vehicleOptions = vehicles.map((vehicle: VehicleModelDataResponse) => {
+          return {name: vehicle.data.attributes.name, value: vehicle.data.id}
```
Comment (ialej001) [inline src/server/router/vehicle.router.ts]: As a suggestion here, in your tRPC query for the models, why not just return `{name: vehicle.data.attributes.year + ' ' + vehicle.data.attributes.name, ... }` in your map function? Or if that don't like that, add a line before your return in the map function `const vehicleYearAndName = [vehicle.data.attributes.year, vehicle.data.attributes.name].join(' ')` and just return `vehicleYearAndName `in the value field for `name`

**Review by ialej001 — CHANGES_REQUESTED** [review]
Overall, I like it and it works. Vehicle models is a bit vague with the apparent duplicates. To clarify, we should include the years in the return object from tRPC. See my comments.

Good work Ticia

File: src/server/router/vehicle.router.ts (hunk 2 above)
Comment (TiciaD) [reply src/server/router/vehicle.router.ts]: Good point added the year to the models listing! I'm going to make a separate ticket for the models data filtering so we can get this merged in sooner

**Review by ialej001 — APPROVED** [review]
Great job :)

File: src/components/Vehicle/VehicleForm.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            />
+          </div>
+          <div className="col-span-2 md:col-span-1">
+            <Select
+              id="distanceUnit"
+              onChange={(e) => setDistanceUnit(e.target.value as DistanceUnit)}
+              value={distanceUnit}
+              color="dark"
+            >
+              <option value="mi">miles</option>
```
Comment (victoriacheng15) [inline src/components/Vehicle/VehicleForm.tsx:160]: I think the default unit should be km?

**Review by victoriacheng15 — COMMENTED** [review]
While you are on this, a friendly reminder, remove the home link from the `travel.tsx`.

Thank you! :D

**Review by amandaalexandre — APPROVED** [review]

**Pull request merged**